### PR TITLE
Swift 4 Update

### PR DESCRIPTION
**Pull request by Narsail**

### CODE REVIEW

**Comment by Narsail** [comment]
@a2 Can you check this? it should be all right to merge and for the Swift 4 Update.

File: MessagePack.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -1013,7 +1014,8 @@
 				PRODUCT_NAME = MessagePack;
 				SDKROOT = macosx;
 				SKIP_INSTALL = YES;
-				SWIFT_VERSION = 3.0;
+				SWIFT_SWIFT3_OBJC_INFERENCE = On;
```
Comment (a2) [inline MessagePack.xcodeproj/project.pbxproj]: Can we remove the Obj-C inference?

File: MessagePack.xcodeproj/xcshareddata/xcschemes/MessagePack watchOS.xcscheme [hunk 2]
```diff
@@ -15,7 +15,7 @@
             <BuildableReference
                BuildableIdentifier = "primary"
                BlueprintIdentifier = "825321D21EFFCAED00914B55"
-               BuildableName = "MessagePack-watchOS.framework"
+               BuildableName = "Messagepack.framework"
```
Comment (a2) [inline MessagePack.xcodeproj/xcshareddata/xcschemes/MessagePack watchOS.xcscheme]: Should this be `MessagePack` with a capital P?

**Review by a2 — COMMENTED** [review]
Just a couple of small things. Looks great otherwise!

File: MessagePack.xcodeproj/project.pbxproj (hunk 1 above)
Comment (Narsail) [reply MessagePack.xcodeproj/project.pbxproj]: Fixed.

File: MessagePack.xcodeproj/xcshareddata/xcschemes/MessagePack watchOS.xcscheme (hunk 2 above)
Comment (Narsail) [reply MessagePack.xcodeproj/xcshareddata/xcschemes/MessagePack watchOS.xcscheme]: Fixed.

**Comment by a2** [comment]
PR looks 👌 (great) now. Waiting for the tests to finish running.

I think this should be version 3.0.0 because it's a breaking change (Swift 4 required). What do you think?

**Comment by Narsail** [comment]
Yup, I agree with you on that. 3.0.0 sounds good. Thanks for your fast response! :)

**Comment by Narsail** [comment]
Can you release it today? The sooner the better for me :)

**Pull request merged**